packages/@aws-cdk/aws-codestarnotifications/lib/rule.ts

```diff
@@ -0,0 +1,262 @@
+   *
+   * @default Status.ENABLED
+   */
+  readonly status?: Status;
+
+  /**
+   * The level of detail to include in the notifications for this resource.
+   * BASIC will include only the contents of the event as it would appear in AWS CloudWatch.
+   * FULL will include any supplemental information provided by AWS CodeStar Notifications and/or the service for the resource for which the notification is created.
+   *
+   * @default DetailType.FULL
+   */
+  readonly detailType?: DetailType;
+
+  /**
+   * The Amazon Resource Name (ARN) of the resource to associate with the notification rule.
+   * Currently, Supported sources include pipelines in AWS CodePipeline and build projects in AWS CodeBuild in this L2 constructor.
+   * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-codestarnotifications-notificationrule.html#cfn-codestarnotifications-notificationrule-resource
+   */
+  readonly source: IRuleSource;
+
+  /**
+   * A list of Amazon Resource Names (ARNs) of Amazon SNS topics and AWS Chatbot clients to associate with the notification rule.
+   * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-codestarnotifications-notificationrule-target.html
+   *
+   * You must specify this property (either via props or via `addTarget`)
+   *
+   * @default - No targets.
+   */
+  readonly targets?: IRuleTarget[];
+
+  /**
+   * A list of event types associated with this notification rule.
+   * For a complete list of event types and IDs, see Notification concepts in the Developer Tools Console User Guide.
+   * @see https://docs.aws.amazon.com/dtconsole/latest/userguide/concepts.html#concepts-api
+   *
+   * You must specify this property (either via props or via `addEvents`)
+   *
+   * @default - No events.
+   */
+  readonly events?: string[];
+}
+
+/**
+ * Represents a notification rule
+ */
+export interface IRule extends IResource {
+
+  /**
+   * The ARN of the notification rule (i.e. arn:aws:codestar-notifications:::notificationrule/01234abcde)
+   * @attribute
+   */
+  readonly ruleArn: string;
+
+  /**
+   * The name of the notification rule
+   * @attribute
+   */
+  readonly ruleName: string;
+}
+
+/**
+ * A new notification rule
+ *
+ * @resource AWS::CodeStarNotifications::NotificationRule
+ */
+export class Rule extends Resource implements IRule {
+  /**
+   * Import an existing notification rule provided an ARN
+   * @param scope The parent creating construct
+   * @param id The construct's name
+   * @param notificationRuleArn Notification rule ARN (i.e. arn:aws:codestar-notifications:::notificationrule/01234abcde)
+   */
+  public static fromNotificationRuleArn(scope: Construct, id: string, notificationRuleArn: string): IRule {
+    const parts = Stack.of(scope).parseArn(notificationRuleArn);
+    class Import extends Resource implements IRule {
+      readonly ruleArn = notificationRuleArn;
+      readonly ruleName = parts.resourceName || '';
```

Comment:
Throw an exception if `resourceName` is empty instead.

Reply:
This ruleName will throw an exception, because the name is not in part of ARN.

Reply:
Sorry, I don't understand what you mean... You have

```ts
    const parts = Stack.of(scope).parseArn(notificationRuleArn);
      readonly ruleName = parts.resourceName || '';
```

So clearly `ruleName` is part of the ARN...?

Reply:
Yes, the ARN like this:
`arn:aws:codestar-notifications:us-east-1:123456789012:notificationrule/af673fcfef0dc2188357ec00876a94d64b777df4`

Reply:
Right. So when a customer calls `Rule. fromNotificationRuleArn()` with a value like `arn:aws:codestar-notifications:us-east-1:123456789012:notificationrule`, you should fail with a descriptive error.

Reply:
Can I do it like this?
```ts
  public static fromNotificationRuleArn(scope: Construct, id: string, notificationRuleArn: string): IRule {
    class Import extends Resource implements IRule {
      readonly ruleArn = notificationRuleArn;
      public get ruleName(): string { throw new Error('cannot retrieve "ruleName" from an imported'); }

      public addTarget(_target?: IRuleTarget) {
        return false;
      }
    }

    return new Import(scope, id);
  }
```

Reply:
No, that doesn't make too much sense.

Do it [like this](https://github.com/aws/aws-cdk/blob/af6d49f2e245b60ae3bbea3bb2c5d283beedba3f/packages/%40aws-cdk/aws-kms/lib/key.ts#L480-L483).

Reply:
> Sorry, I don't understand what you mean... You have
> 
> ```ts
>     const parts = Stack.of(scope).parseArn(notificationRuleArn);
>       readonly ruleName = parts.resourceName || '';
> ```
> 
> So clearly `ruleName` is part of the ARN...?

Oops! I'm sorry that I misread your reply.
Actually, `ruleName` is `not` part of the ARN.

The real ARN of CodeStar Notification Rule like this: 
```
arn:aws:codestar-notifications:us-east-1:123456789012:notificationrule/af673fcfef0dc2188357ec00876a94d64b777df4
```

If I set `test-notification` to `ruleName` of `RuleProps` the name which is `test-notification` is not in the ARN.

As you can see, `test-notification` I named is not in ARN `arn:aws:codestar-notifications:us-east-1:123456789012:notificationrule/af673fcfef0dc2188357ec00876a94d64b777df4`

Reply:
> No, that doesn't make too much sense.
> 
> Do it [like this](https://github.com/aws/aws-cdk/blob/af6d49f2e245b60ae3bbea3bb2c5d283beedba3f/packages/%40aws-cdk/aws-kms/lib/key.ts#L480-L483).

Is it still a solution?

Reply:
I think this means you should remove the `ruleName` property from `IRule` completely, and only have `ruleArn` there.

Reply:
Ok, it will be adjust together with renaming.

Reply:
Removed.